### ISSUE TITLE
False positive with contravariant type variables in overloads implemented on `Protocol`s

### ISSUE THREAD

**Issue by bzoracler** [python/mypy#20108]
**Bug Report, To Reproduce, & Actual Behaviour**

See [mypy Playground](https://mypy-play.net/?mypy=1.18.2&python=3.12&gist=bf7a837f7edce31c440839c0b7889352):

```python
from typing import TypeVar, Protocol, overload

_T1_contra = TypeVar("_T1_contra", contravariant=True)
_T2_contra = TypeVar("_T2_contra", contravariant=True)

class A(Protocol[_T1_contra, _T2_contra]):  # E: Contravariant type variable "_T2_contra" used in protocol where covariant one is expected  [misc]
    @overload
    def method(self, a: _T1_contra) -> None: ...
    @overload
    def method(self, a: _T2_contra) -> None: ...
    def method(self, a: object) -> None: ...
```

No errors are emitted if type parameter syntax is used (which should automatically imply contravariance), or if `Protocol` is replaced by `Generic`.

**Expected Behavior**

No errors.

**Your Environment**

- Mypy version used: 1.18.2
